sdks/python/apache_beam/runners/dataflow/internal/apiclient.py

```diff
@@ -301,6 +351,22 @@ def __init__(self, packages, options, environment_version, pipeline_url):
           dataflow.Environment.SdkPipelineOptionsValue.AdditionalProperty(
               key='display_data', value=to_json_value(items)))
 
+  def _get_environments_from_tranforms(self):
```

Comment:
Any reason to not just `return self._pipeline_proto.environments.values()`? Do we expect unused environments to be there for any reason?